ci/docker/install/ubuntu_publish.sh

```diff
@@ -57,14 +57,33 @@ mv apache-maven-3.3.9/ /usr/local/maven/
 update-alternatives --install /usr/bin/mvn mvn /usr/local/maven/apache-maven-3.3.9/bin/mvn 1
 update-ca-certificates -f
 
-apt-get install -y python python3
+# patchelf available starting Ubuntu 16.04; compile from source for Ubuntu 14.04
+mkdir /usr/local/patchelf
+cd /usr/local/patchelf
+curl -L -o patchelf-0.10.tar.gz https://github.com/NixOS/patchelf/archive/0.10.tar.gz
+tar xzf patchelf-0.10.tar.gz
+cd /usr/local/patchelf/patchelf-0.10
+./bootstrap.sh
+./configure
+make
+sudo make install
+cd /
+
+apt-get install -y python python-pip python3 python3-pip
 
 # the version of the pip shipped with ubuntu may be too lower, install a recent version here
-wget -nv https://bootstrap.pypa.io/get-pip.py
-python3 get-pip.py
-python2 get-pip.py
+# Restrict pip version to <19 due to use of Python 3.4 on Ubuntu 14.04
+python2 -m pip install --upgrade 'pip<19'
+python3 -m pip install --upgrade 'pip<19'
 
-apt-get remove -y python3-urllib3
+# Restrict numpy version to <1.18 due to use of Python 3.4 on Ubuntu 14.04
+python2 -m pip install --upgrade --ignore-installed nose cpplint==1.3.0 'numpy>1.16.0,<1.17' nose-timer 'requests<2.19.0,>=2.18.4' h5py==2.8.0rc1 scipy==1.0.1 boto3
+python3 -m pip install --upgrade --ignore-installed nose cpplint==1.3.0 pylint==2.3.1 'numpy>1.16.0,<1.18' nose-timer 'requests<2.19.0,>=2.18.4' h5py==2.8.0rc1 scipy==1.0.1 boto3
 
-pip2 install nose cpplint==1.3.0 'numpy>1.16.0,<2.0.0' nose-timer 'requests<2.19.0,>=2.18.4' h5py==2.8.0rc1 scipy==1.0.1 boto3
-pip3 install nose cpplint==1.3.0 pylint==2.3.1 'numpy>1.16.0,<2.0.0' nose-timer 'requests<2.19.0,>=2.18.4' h5py==2.8.0rc1 scipy==1.0.1 boto3
+# CMake 3.13.2+ is required
+mkdir /opt/cmake && cd /opt/cmake
+wget -nv https://cmake.org/files/v3.13/cmake-3.13.5-Linux-x86_64.sh
+sh cmake-3.13.5-Linux-x86_64.sh --prefix=/opt/cmake --skip-license
+ln -s /opt/cmake/bin/cmake /usr/local/bin/cmake
+rm cmake-3.13.5-Linux-x86_64.sh
```

Comment:
Don't we already install cmake somewhere else?

Reply:
Not on the machines testing static build pipeline. It's installed in ubuntu_core.sh, but that's not executed for static build pipeline.

Reply:
Ubuntu 14.04 is used for static build

Reply:
Shall we maybe extract that part into an own script so we can install the same cmake version for publishing as well as ci?

Reply:
I think it's sensible to separate Ubuntu 14.04 and "recent" Ubuntu code. Many workarounds are needed on Ubuntu 14.04, see for example https://github.com/apache/incubator-mxnet/pull/17474

Reply:
Right, but shall we try to aim to make the two environments as similar as possible? I think we want to avoid running into issues caused by diverging environment

Reply:
I agree, let's tackle it in a separate PR though once the CI runs stable again. (Refactoring the CI build scripts may introduce bugs at first and it's hard to find them if a lot of pipelines fail randomly as currently)